src/figuresClasses.ts

```diff
@@ -1,19 +1,90 @@
 export interface Figure {
-
+  color: string;
+  shape: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  color: string;
+
+  shape: string;
+
+  sides: number[];
+
+  constructor(color: string, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Length of each side must be greater than 0.');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('The sides cannot form a triangle.');
+    }
```

Comment:
would be nice to move to a separate method (like `validateTriangle`)

Reply:
made separate funcs